ansible_rulebook/builtin.py

```diff
@@ -815,3 +823,15 @@ def _get_events(variables: Dict):
     elif "events" in variables:
         return variables["events"]
     return {}
+
+
+def _collect_extra_vars(variables: Dict, user_extra_vars: Dict, facts: Dict):
```

Comment:
If I have understood well this function, if an environment variable name matches an user-extra-variable name, now the extra_var it would be replaced by the environment variable, but extra-vars should take precedence. Am I wrong?  If the user wants to override this behavior should use the environment variable in the extra_vars declaration. Such as:

```yaml
    - name: r1
      condition: event.status == "OK"
      action:
        run_playbook:
          name: test.yml
          extra_vars:
            project_name: "{{ project_name }}" # this is the project_name taken from environment variables instead of extra-vars file. 
```

What do you think?  @mkanoor @bzwei

Reply:
@Alex-Izquierdo Both the variables and the env vars all go into the variables pool here and can be used for substitution. 
https://github.com/ansible/ansible-rulebook/blob/38f3bab204ad12234705c06d532ca1630c3b45e1/ansible_rulebook/app.py#L126

Reply:
Yeah, but what if there is a name collision? which takes precedence? We must document it.

Reply:
Sure. Will include in document

Reply:
Document added.